app/models/ticket_purchase.rb

```diff
@@ -58,4 +63,24 @@ def self.update_quantity(conference, quantity, ticket, user)
     purchase.quantity = quantity if quantity > 0
     purchase
   end
+
+  def pay(payment)
+    self.update_attributes(paid: true, payment_id: payment.id)
```

Comment:
you can write this as:

``` Ruby
update_attributes(paid: true, payment: payment)
```

`self` is not needed. Please read: https://hackhands.com/three-golden-rules-understand-self-ruby :wink: And Rubocop can also help you, as the `Style/RedundantSelf` cop is enabled. 

And as we have a relation, you can use it instead of the `id` :slightly_smiling_face: